src/components/encoding-pane/filter-pane.tsx

```diff
@@ -0,0 +1,226 @@
+import {ExpandedType} from 'compassql/build/src/query/expandedtype';
+import {Schema} from 'compassql/build/src/schema';
+import {isWildcard} from 'compassql/build/src/wildcard';
+import * as React from 'react';
+import * as CSSModules from 'react-css-modules';
+import {ConnectDropTarget, DropTarget, DropTargetCollector, DropTargetSpec} from 'react-dnd';
+import {OneOfFilter, RangeFilter} from 'vega-lite/build/src/filter';
+import {TimeUnit} from 'vega-lite/build/src/timeunit';
+import {FILTER_ADD, FILTER_MODIFY_TIME_UNIT,
+  FILTER_REMOVE, FilterAction} from '../../actions/filter';
+import {DraggableType} from '../../constants';
+import {ShelfFieldDef} from '../../models/shelf/encoding';
+import {getDefaultList, getDefaultRange} from '../../reducers/shelf/filter';
+import {DraggedFieldIdentifier} from '../field';
+import {Field} from '../field/index';
+import * as styles from './filter-pane.scss';
+import {FunctionPicker} from './function-picker';
+import {OneOfFilterShelf} from './one-of-filter-shelf';
+import {RangeFilterShelf} from './range-filter-shelf';
+
+
+
+/**
+ * Props for react-dnd of FilterShelf
+ */
+export interface FilterPaneDropTargetProps {
+  connectDropTarget: ConnectDropTarget;
+
+  isOver: boolean;
+
+  item: Object;
+}
+
+export interface FilterPanePropsBase {
+  filters: Array<RangeFilter | OneOfFilter>;
+  fieldDefs: ShelfFieldDef[];
+  schema: Schema;
+  handleAction?: (action: FilterAction) => void;
+}
+
+interface FilterPaneProps extends FilterPaneDropTargetProps, FilterPanePropsBase {};
+
+class FilterPaneBase extends React.Component<FilterPaneProps, {}> {
+  // private index: number;
+
+  public constructor(props: FilterPaneProps) {
+    super(props);
+    this.filterModifyTimeUnit = this.filterModifyTimeUnit.bind(this);
+  }
+
+  public render() {
+    const {filters, connectDropTarget} = this.props;
+    const filterShelves = filters.map((filter, index) => {
+      return this.renderFilterShelf(filter, index);
+    });
+    return connectDropTarget(
+      <div>
+        {filterShelves}
+        {this.fieldPlaceholder()}
+      </div>
+    );
+  }
+
+  protected filterRemove(index: number) {
+    const {handleAction} = this.props;
+    handleAction({
+      type: FILTER_REMOVE,
+      payload: {
+        index: index
+      }
+    });
+  }
+
+  protected filterModifyTimeUnit(timeUnit: string, index: number) {
+    const {handleAction} = this.props;
+    if (timeUnit === '-') {
+      timeUnit = undefined;
+    }
+
+    handleAction({
+      type: FILTER_MODIFY_TIME_UNIT,
+      payload: {
+        index: index,
+        timeUnit: timeUnit as TimeUnit
+      }
+    });
+  }
+
+  private renderFilterShelf(filter: RangeFilter | OneOfFilter, index: number) {
+    const {fieldDefs, schema, handleAction} = this.props;
+    const fieldIndex = schema.fieldNames().indexOf(filter.field);
+    const fieldDef = fieldDefs[fieldIndex];
+    let domain = schema.domain(fieldDef as FieldQuery);
```

Comment:
Seeing this line, I assume this PR is not ready?  (I remember helping you a bit on this this morning.)